darwin/modules/homebrew.nix

```diff
@@ -1,11 +1,29 @@
-{ config, lib, ... }:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  configuredCasks = builtins.toFile "new-casks" (
+    lib.concatMapStringsSep "\n" (cask: cask.name) config.homebrew.casks
+  );
+  cleanupCasks = pkgs.writeShellScript "cleanup-casks" ''
+    set -eu -o pipefail
+    comm -23 \
+      <(/opt/homebrew/bin/brew list --casks -q -1 | sort -u) \
+      <(sort -u ${configuredCasks}) | xargs --no-run-if-empty -I {} /opt/homebrew/bin/brew uninstall --cask {}
+  '';
+in
 {
   homebrew.enable = true;
   system.activationScripts.homebrew.text = lib.mkIf config.homebrew.enable (
     lib.mkBefore ''
       if [[ ! -f "${config.homebrew.brewPrefix}/brew" ]]; then
         /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
       fi
+
+      ${cleanupCasks}
```

Comment:
@emilazy I guess this would be a nice optional addition to darwin-nix?

Reply:
Does this differ from `homebrew.onActivation.cleanup = "uninstall";`?